cli/azd/cmd/init.go

```diff
@@ -78,6 +78,13 @@ func (i *initFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOpt
 		"",
 		"Name or ID of an Azure subscription to use for the new environment",
 	)
+	local.BoolVarP(
+		&i.fromCode,
+		"from-code",
+		"",
+		false,
+		"Initializes a new application from your existing code.",
+	)
```

Comment:
Should we use the pattern:

```go
f := local.VarPF(...)
// ensure the flag behaves as a common boolean flag which is set to true when used without any other arg
f.NoOptDefVal = "true"
```

That's what we do for a few bool args around there, like `use-device-code`

Reply:
I'm not 100% sure what that pattern buys us?   Is that just about configuring the default when it is not on the command line? If we wanted to default the value to true, I think we would set the `false` to `true` on L85.

FWIW, I prefer what @weikanglim has here if they are semantically the same.

Reply:
>  I think we would set the false to true on L85.

That would mean that if you don't set the flag, at all, you will get `true`, which is not what we want, @ellismg 

IIRC, what we get from that pattern is the ability to know if the flag is false b/c it was not manually added, or if the flag was added like `azd init --from-code=false`.

Reply:
The `boolPtr` pattern is documented as follows:

> // boolPtr implements a pflag.Value and allows us to distinguish between a flag value being explicitly set to
// bool vs not being present.

So, we'd adopt the pattern if we need to distinguish between `--from-code 0` versus unset. In `auth login`, we want:

```bash
$ azd auth login # unset, but can be autodetermined by azd
$ azd auth login --use-device-code=false # explicitly set, azd will not modify behavior
```

 but we don't need that here, is what I'm thinking.

Reply:
> but we don't need that here

Shouldn't we collect telemetry usage for all invocations?  Maybe we will wonder in the future, how many folks are passing bool flags explicitly v/s letting azd to set defaults. :)